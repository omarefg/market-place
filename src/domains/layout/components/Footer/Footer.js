import React from "react";
import classes from "./Footer.module.css";

export function Footer(props) {
  const { children } = props;
  return (
    <footer data-testid="Footer_root" className={classes.footer}>
      {children}
    </footer>
  );
}
