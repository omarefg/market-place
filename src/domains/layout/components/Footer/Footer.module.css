.footer {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 20px 40px;
  background-color: var(--color-2);
  position: fixed;
  bottom: 0;
  right: 0;
  left: 0;
  height: 54px;
}
