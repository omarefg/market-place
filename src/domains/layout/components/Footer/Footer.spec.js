import { render } from "@testing-library/react";
import { Footer } from "./Footer";

describe("@domains/layout/components/Footer", () => {
  let wrapper, defaultProps;
  beforeEach(() => {
    defaultProps = {
      children: <div data-testid="Footer_children" />,
    };
    wrapper = (props = {}) => {
      const finalProps = { ...defaultProps, ...props };
      return render(<Footer {...finalProps} />);
    };
  });
  afterEach(() => {});
  it("Given a valid call it must render without crashing", () => {
    // Arrange
    const { container } = wrapper();
    //Assert
    expect(container).toMatchSnapshot();
  });

  it("Given valid children it must render that node", () => {
    // Arrange
    const { getByTestId } = wrapper();
    const children = getByTestId("Footer_children");
    const root = getByTestId("Footer_root");
    // Act
    const renderChild = children.parentNode.isEqualNode(root);
    // Assert
    expect(renderChild).toBe(true);
  });
});
