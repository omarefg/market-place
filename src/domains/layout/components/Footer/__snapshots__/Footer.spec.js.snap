// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@domains/layout/components/Footer Given a valid call it must render without crashing 1`] = `
<div>
  <footer
    class="footer"
    data-testid="Footer_root"
  >
    <div
      data-testid="Footer_children"
    />
  </footer>
</div>
`;
